### DISCUSSION TITLE
How to use a different backend betwen persistence and queue ?

### DISCUSSION THREAD

**Discussion by yannrouillard** [Netflix/conductor#2531] [Q&A]
Hi,

I saw some discussion talking about using one backend for persistence and another for queue, like this one: https://github.com/Netflix/conductor/discussions/2372#discussioncomment-1058751
or here: https://netflix.github.io/conductor/extend/ where they talk about different implementations for QueueDao, ExecutionDao...

However it is not clear to me how we can configure that.
We would like to use MySQL for persistence and Redis for the queue.

Can someone enlighten me ?

Thanks in advance !

Yann

**Comment by yannrouillard** [comment]
@aravindanr since you mentioned the fact that you used different backend for persistence and queue in https://github.com/Netflix/conductor/discussions/2372#discussioncomment-1058751, can I ask you how you configure that in Conductor ?

**Comment by yannrouillard** [comment]
I think I have an partial answer in this discussion: https://github.com/Netflix/conductor/discussions/2224
This is not possible currently but there is a community effort toward that in this discussion.

**Comment by buzz08** [comment]
Hey @yannrouillard - This is something we want too. It should be possible to implement this. I am curious if you are willing to collaborate. cc: @v1r3n

**Comment by aravindanr** [comment]
@yannrouillard We have different implementations of `ExecutionDAO` and `QueueDAO` in the class path. They are selectively enabled/disabled using `@ConditionalOnProperty` supported by Spring Boot. For instance, we have an implementation of `ExecutionDAO` from the `cassandra-persistence` module and our own implementation of `QueueDAO` backed by Redis. These DAO implementations are enabled/disabled using a property.

Note that we wrap conductor as a library in our own application, which enables us to do this sort of customization. If you are running Conductor on its own, the options are limited and #2224 is an effort to address that. But it has been dormant for a while.
